datafusion/optimizer/src/simplify_expressions.rs

```diff
@@ -2553,4 +2571,56 @@ mod tests {
 
         assert_optimized_plan_eq(&plan, expected);
     }
+
+    #[test]
+    fn simplify_expr_for_constant_fold_test() {
+        let schema = DFSchema::new_with_metadata(
+            vec![DFField::new(None, "x", DataType::Int32, false)],
+            HashMap::new(),
+        )
+        .unwrap();
+
+        // x + (1 + 3)
+        let expr = Expr::BinaryExpr {
+            left: Box::new(col("x")),
+            op: Operator::Plus,
+            right: Box::new(Expr::BinaryExpr {
+                left: Box::new(lit(1)),
+                op: Operator::Plus,
+                right: Box::new(lit(3)),
+            }),
+        };
```

Comment:
These expressions can be built in a much less verbose way:

```suggestion
        let expr = col("x") + (lit(1) + lit(3));
```